packages/nextjs/src/utils/handlers.ts

```diff
@@ -0,0 +1,23 @@
+import { captureException, flush, Handlers, withScope } from '@sentry/node';
+import { NextApiRequest, NextApiResponse } from 'next';
+
+const { parseRequest } = Handlers;
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const withSentry = (handler: (req: NextApiRequest, res: NextApiResponse) => Promise<void>) => {
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+  return async (req: NextApiRequest, res: NextApiResponse) => {
+    try {
+      // TODO: Start Transaction
+      return await handler(req, res); // Call Handler
+      // TODO: Finish Transaction
+    } catch (e) {
+      withScope(scope => {
+        scope.addEventProcessor(event => parseRequest(event, req));
+        captureException(e);
+      });
+      await flush(2000);
```

Comment:
Is flushing on every error required? https://github.com/getsentry/sentry-javascript/pull/3301#discussion_r614928419

Reply:
@iker-barriocanal @lobsterkatie i've found that flushing here breaks Next.js API Routes in dev mode when using `cors` middleware.

It results in an `ERR_STREAM_WRITE_AFTER_END` error.

Removing the call to flush resolves it, but i'm not sure what the side affects of doing this might be.

Reply:
We flush to ensure that when API routes are deployed as serverless functions, the event gets sent before the function ends. @ashconnell - can you share a sample API route which is crashing for you?

Reply:
@lobsterkatie here's a reproduction. 

I'm using the latest Next.js and Sentry versions. 
To reproduce the error you need to fetch the api route from another origin so that it triggers the cors preflight etc.

```js
import { withSentry } from '@sentry/nextjs'
import Cors from 'cors'

// This is straight from Next.js docs on how to use cors
// See: https://nextjs.org/docs/api-routes/api-middlewares#connectexpress-middleware-support
function runMiddleware(req, res, fn) {
  return new Promise((resolve, reject) => {
    fn(req, res, result => {
      if (result instanceof Error) {
        return reject(result)
      }

      return resolve(result)
    })
  })
}

const cors = async (req, res) => {
  return await runMiddleware(req, res, Cors({}))
}

async function handler(req, res) {
  await cors(req, res)
  res.status(200).send({ message: 'Hello!' })
}

export default withSentry(handler)
```

Reply:
I may have found something. So when @sentry/nextjs calls flush, @sentry/node attempts to get the client which does not exist, and rejects the promise with `false`

```js
// node/sdk.ts
export async function flush(timeout?: number): Promise<boolean> {
  const client = getCurrentHub().getClient<NodeClient>();
  if (client) {
    return client.flush(timeout);
  }
  return Promise.reject(false);
}
```

Since the @sentry/nextjs flush is awaiting flush, the whole `withSentry` call is rejected, which Next.js catches and calls `res.end()`.